src/jit/emitxarch.cpp

```diff
@@ -4088,6 +4088,119 @@ void emitter::emitIns_R_A_I(instruction ins, emitAttr attr, regNumber reg1, GenT
     emitCurIGsize += sz;
 }
 
+//------------------------------------------------------------------------
+// emitIns_R_RM_I: emits the code for an instruction that takes 3 operands:
+//                 a register operand, an operand that may be memory or register and an immediate
+//                 and that returns a value in register
+//
+// Arguments:
+//    ins       -- The instruction being emitted
+//    attr      -- The emit attribute
+//    reg1      -- The first operand, a register
+//    rmOp      -- The second operand, which may be a memory node or a node producing a register
+//    ival      -- The immediate operand
+//
+void emitter::emitIns_R_RM_I(instruction ins, emitAttr attr, regNumber reg1, GenTree* rmOp, int ival)
```

Comment:
I wouldn't put this in the emitter. Ideally the emitter interface should not deal with `GenTree`, only instructions, registers and immediates/displacements.

Reply:
You could probably just update `genHWIntrinsic_R_RM_I()`: https://github.com/dotnet/coreclr/blob/407cc7d69c1db869e2f4418e180f51394fa3c73d/src/jit/hwintrinsiccodegenxarch.cpp#L528

Reply:
Hmm, it actually looks like most of this logic was pulled from the previous impl of `genHWIntrinsic_R_RM_I`

Reply:
I think most of the `genHWIntrinsic_R_...` methods are probably more "general-purpose" than HWIntrinsic specific.

That is, outside some asserts and the explicit code path for `containedOp->OperIsHWIntrinsic()`, they are just taking a node and determining which emitter call to make (`R`, `AR`, `C`, `A`, `S`, etc).

It might be worth refactoring all of these to be more generally usable?

Reply:
The various "RM" handling functions and the relevant emitter interface functions are pretty messed up as is. It will take  a significant effort to address all that and it certainly doesn't belong in this PR. For now juts avoiding adding more stuff to the emitter interface should be enough. It already suffers from duplication, bad function and parameter names etc.

Reply:
> It will take a significant effort to address all that and it certainly doesn't belong in this PR.

Completely right, I was more asking as a potential future work item.

Reply:
As it turns out, extracting the HWIntrinsic-dependencies from this was a bit problematic, so although I moved to to `CodeGen::inst_RV_TT_IV()`, it is still under `#ifdef FEATURE_HW_INTRINSICS`.